src/util/mpir_hwtopo.c

```diff
@@ -482,10 +482,12 @@ int MPIR_hwtopo_mem_bind(void *baseaddr, size_t len, MPIR_hwtopo_gid_t gid)
 #ifdef HAVE_HWLOC
     const struct hwloc_topology_support *support = hwloc_topology_get_support(hwloc_topology);
     if (!support->membind->set_area_membind) {
-        fprintf(stderr,
-                "%s: hwloc_set_area_membind() is not supported, skipping memory binding\n",
-                __func__);
-        return MPI_ERR_OTHER;
+#ifdef HAVE_ERROR_CHECKING
+        ret =
+            MPIR_Err_create_code(MPI_SUCCESS, MPIR_ERR_RECOVERABLE, __func__, __LINE__,
+                                 MPI_ERR_OTHER, "**nomembind", 0);
```

Comment:
nitpick: the error class is `MPI_ERR_OTHER`. This commit is adding error codes and names.

Reply:
I will change the commit message